### PR TITLE
Add superAdmin validation for patch and delete operations

### PR DESCRIPTION
**Pull request by JeethJJ**
With this update superAdmin related patch and delete scim operations will only be allowed to be performed by the superAdmin themselves.

### CODE REVIEW

**Comment by Thumimku** [comment]
Is there any issue related to this fix?

File: components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/resources/UserResource.java [hunk 1]
```diff
@@ -145,6 +147,12 @@ public Response deleteUser(@PathParam(SCIMProviderConstants.ID) String id,
             // obtain the user store manager
             UserManager userManager = IdentitySCIMManager.getInstance().getUserManager();
 
+            String superAdminID = AdminAttributeUtil.getSuperAdminID();
+            String loggedInUser = SCIMCommonUtils.getLoggedInUserID();
+            if ((superAdminID.equals(id)) && (!loggedInUser.equals(id))) {
+                return Response.status(Response.Status.FORBIDDEN).build();
```
Comment (Thumimku) [inline components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/resources/UserResource.java:153]: let's add a debug log here

File: components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/resources/UserResource.java [hunk 2]
```diff
@@ -145,6 +147,12 @@ public Response deleteUser(@PathParam(SCIMProviderConstants.ID) String id,
             // obtain the user store manager
             UserManager userManager = IdentitySCIMManager.getInstance().getUserManager();
 
+            String superAdminID = AdminAttributeUtil.getSuperAdminID();
+            String loggedInUser = SCIMCommonUtils.getLoggedInUserID();
+            if ((superAdminID.equals(id)) && (!loggedInUser.equals(id))) {
```
Comment (Thumimku) [inline components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/resources/UserResource.java:152]: what if logged in user has required role to perform delete action? 
why this delete operation restricted to super admin user?

Comment (JeethJJ) [reply components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/resources/UserResource.java:152]: The operation to delete the SuperAdmin is only limited to the SuperAdmin themselves. No other user should be allowed to perform this action.

File: components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/resources/UserResource.java (hunk 1 above)
Comment (JeethJJ) [reply components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/resources/UserResource.java:160]: Ack & done

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/AdminAttributeUtil.java [hunk 3]
```diff
@@ -183,6 +190,34 @@ private static String getAdminGroupName(String adminRoleName, String domainName)
...
+            RealmConfiguration realmConfig = userRealm.getRealmConfiguration();
+            String adminUser = realmConfig.getAdminUserName();
+
+            UserIDResolver userIDResolver = new UserIDResolver();
+            String adminUserID = userIDResolver.getIDByName(adminUser, tenantDomain);
+
+            return adminUserID;
+
+        } catch (org.wso2.carbon.user.api.UserStoreException e) {
+            String error = "Error obtaining user realm.";
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/AdminAttributeUtil.java:214]: Can get rid of this variable.

**Pull request merged**